questionpy_sdk/webserver/question_ui.py

```diff
@@ -0,0 +1,622 @@
+            index -= val[i]
+        i += 1
+    return roman_num
+
+
+def replace_shuffled_indices(element: etree._Element, index: int) -> None:
+    for index_element in is_iterable(element.xpath(".//qpy:shuffled-index",
+                                                   namespaces={'qpy': "http://questionpy.org/ns/question"})):
+        format_style = index_element.get("format", "123")
+
+        if format_style == "123":
+            index_str = str(index)
+        elif format_style == "abc":
+            index_str = int_to_letter(index).lower()
+        elif format_style == "ABC":
+            index_str = int_to_letter(index).upper()
+        elif format_style == "iii":
+            index_str = int_to_roman(index).lower()
+        elif format_style == "III":
+            index_str = int_to_roman(index).upper()
+        else:
+            index_str = str(index)
+
+        # Replace the index element with the new index string
+        new_text_node = etree.Element("span")  # Using span to replace the custom element
+        new_text_node.text = index_str
+
+        if index_element.tail:
+            new_text_node.tail = index_element.tail
+
+        index_element.getparent().replace(index_element, new_text_node)
+
+
+class QuestionUIRenderer:
+    XHTML_NAMESPACE: str = "http://www.w3.org/1999/xhtml"
+    QPY_NAMESPACE: str = "http://questionpy.org/ns/question"
+    question: etree._Element
+    placeholders: dict[str, str]
+
+    def __init__(self, xml: str, placeholders: dict[str, str], seed: Optional[int] = None) -> None:
+        self.seed = seed
+        self.xml = xml
+        self.placeholders = placeholders
+        self.question = etree.fromstring(xml.encode())
+
+    def get_metadata(self) -> QuestionMetadata:
+        """Extracts metadata from the question UI.
+
+        Returns:
+            QuestionMetadata: question_metadata
+        """
+        question_metadata = QuestionMetadata()
+        namespaces: dict[str, str] = {'xhtml': self.XHTML_NAMESPACE, 'qpy': self.QPY_NAMESPACE}
+
+        # Extract correct responses
+        for element in self.question.findall(".//qpy:formulation//*[@qpy:correct-response]",
+                                             namespaces=namespaces):
+            name = element.get("name")
+            if not name:
+                continue
+
+            if element.tag.endswith("input") and element.get("type") == "radio":
+                value = element.get("value")
+            else:
+                value = element.get(f"{{{self.QPY_NAMESPACE}}}correct-response")
+
+            if not value:
+                continue
+
+            question_metadata.correct_response[name] = value
+
+        # Extract other metadata
+        for element_type in ['input', 'select', 'textarea', 'button']:
+            for element in self.question.findall(f".//qpy:formulation//{{{self.XHTML_NAMESPACE}}}{element_type}",
+                                                 namespaces=namespaces):
```

Comment:
Tut es hier nicht auch 
```suggestion
            for element in self.question.findall(f".//qpy:formulation//xhtml:{element_type}",
                                                 namespaces=namespaces):
```